### ISSUE TITLE
Slightly misleading documentation for read_events

### ISSUE THREAD

**Issue by ousado** [inotify-rs/inotify#102]
[They state that](https://docs.rs/inotify/0.5.1/inotify/struct.Inotify.html#method.read_events) "This function directly returns all errors from the call to read. " - however EAGAIN/EWOULDBLOCK is signaled by returning an empty slice.

**Comment by hannobraun** [comment]
Good catch, thank you! I'll correct this the next time I work on this project (which might take a good number of months). If someone wants to create a pull request in the meantime, that would be very appreciated.

**Comment by zacheryph** [comment]
I would not consider this a problem or bad documentation?  The underlying `EAGAIN` and `EWOULDBLOCK` are error _codes_, but I would not call them error _conditions_.

As well, the documentation does already clearly state that an empty iterator is returned if nothing is available. It also says to use `read_events_blocking` if you want at least one result. `EAGAIN` and `EWOULDBLOCK` say "nothing is available without blocking". There is in fact no _error_ condition though.

**Comment by hannobraun** [comment]
@zacheryph I think you're right, strictly speaking, but in my experience, it usually helps if documentation is as explicit as possible. I don't consider this issue a big deal at all, but I think it would be an improvement if that one sentence were changed.
